File: LeetCode/1342. Number of Steps to Reduce a Number to Zero/solution.cpp
/**
 * 204 / 204 test cases passed.
 * Runtime: 0 ms
 * Memory Usage: 5.9 MB 
 */
class Solution {
public:
    int numberOfSteps(int num) {
        int step = 0;
        while (num > 0) {
            if (num & 1) --num;
            else num >>= 1;
            ++step;
        }
        return step;
    }
};

File: LeetCode/1342. Number of Steps to Reduce a Number to Zero/solution.py
"""
204 / 204 test cases passed.
Runtime: 40 ms
Memory Usage: 14.8 MB
"""
class Solution:
    def numberOfSteps(self, num: int) -> int:
        step = 0
        while num > 0:
            num = num - 1 if num & 1 else num >> 1
            step += 1
        return step
